src/components/frame/v5/pages/BalancePage/partials/BalanceTable/BalanceTable.tsx

```diff
@@ -53,16 +82,94 @@ const BalanceTable: FC<BalanceTableProps> = ({ data }) => {
     nativeTokenStatus,
     Number(selectedDomain?.nativeId) || undefined,
   );
-  const { getMenuProps } = useGetTableMenuProps(
-    data,
-    toggleAddFundsModalOn,
-    nativeTokenStatus,
-    nativeToken,
-  );
+  const getMenuProps = ({ index }) => {
+    const selectedTokenData = data[index]?.token;
+    const isTokenNative =
+      selectedTokenData?.tokenAddress === nativeToken?.tokenAddress;
+
+    return {
+      cardClassName: 'min-w-[9.625rem] whitespace-nowrap',
+      items: [
+        ...(!isTokenNative && !nativeTokenStatus?.unlocked
+          ? [
+              {
+                key: 'add_funds',
+                onClick: () => {
+                  toggleAddFundsModalOn();
+                },
+                label: formatMessage(MSG.labelAddFunds),
+                icon: 'add',
+              },
+            ]
+          : []),
+        {
+          key: 'view_ethscan',
+          renderItemWrapper: (props, children) => (
+            <Link
+              to={getBlockExplorerLink({
+                linkType: 'address',
+                addressOrHash: selectedTokenData?.tokenAddress || '',
+              })}
+              {...props}
+            >
+              {children}
+            </Link>
+          ),
+          label: formatText(
+            { id: 'balancePage.labelEthscan.viewOn' },
+            {
+              networkName: DEFAULT_NETWORK_INFO.blockExplorerName,
+            },
+          ),
+          icon: 'arrow-square-out',
+        },
+        ...(isTokenNative
+          ? [
+              {
+                key: 'mint_tokens',
+                onClick: () => {
+                  toggleActionSidebarOn({
+                    [ACTION_TYPE_FIELD_NAME]: ACTION.MINT_TOKENS,
+                  });
+                },
+                label: formatMessage(MSG.labelMintToken),
+                icon: 'bank',
+              },
+            ]
+          : []),
+        {
+          key: 'transfer_funds',
```

Comment:
Didn't we have a issue that fixed the "string" key names ?

Reply:
I also remember at some point seeing an issue/PR for that but it seems like those changes are not on `master` yet